community/php7/APKBUILD

```diff
@@ -157,203 +322,285 @@ build() {
-	mv "$pkgdir"/usr/sbin "$subpkgdir"/usr/ || return 1
-	mv "$pkgdir"/etc/$pkgname/php-fpm* "$subpkgdir"/etc/$pkgname/ || return 1
+	_mv var "$subpkgdir"/
+	_mv usr/share/$pkgname/fpm "$subpkgdir"/var/lib/$pkgname/
+	_mv usr/sbin "$subpkgdir"/usr/
+	_mv etc/$pkgname/php-fpm* "$subpkgdir"/etc/$pkgname/
 
 	local file; for file in php-fpm.conf php-fpm.d/www.conf; do
 		mv "$subpkgdir"/etc/$pkgname/$file.default \
-			"$subpkgdir"/etc/$pkgname/$file || return 1
+			"$subpkgdir"/etc/$pkgname/$file
 	done
 
-	install -D -m755 "$srcdir"/$pkgname-fpm.initd "$subpkgdir"/etc/init.d/php-fpm7 || return 1
-	install -D -m644 "$srcdir"/$pkgname-fpm.logrotate "$subpkgdir"/etc/logrotate.d/php-fpm7
+	install -D -m 755 "$srcdir"/$pkgname-fpm.initd \
+		"$subpkgdir"/etc/init.d/php-fpm$_suffix
+
+	install -D -m 644 "$srcdir"/$pkgname-fpm.logrotate \
+		"$subpkgdir"/etc/logrotate.d/php-fpm$_suffix
+
+	mkdir -p "$subpkgdir"/var/log/$pkgname
 }
 
 pear() {
-	pkgdesc="PHP Extension and Application Repository"
+	pkgdesc="PHP$_suffix Extension and Application Repository"
 	depends="$pkgname $pkgname-xml"
 
-	mkdir -p "$subpkgdir"/usr/bin \
-		"$subpkgdir"/etc/$pkgname
+	cd "$pkgdir"
+
+	# pecl needs xml extension and since we build it as shared, it must be
+	# explicitly declared to be loaded.
+	sed -i 's/\$INCARG/& -d extension=xml.so/' usr/bin/pecl
 
-	local file; for file in pecl pear; do
-		mv "$pkgdir"/usr/bin/$file "$subpkgdir"/usr/bin/
+	mkdir -p "$subpkgdir"/usr/bin
+	local file; for file in pecl pear peardev; do
+		mv usr/bin/$file "$subpkgdir"/usr/bin/$file$_suffix
+		[ "$_default_php" = yes ] \
+			&& ln -s $file$_suffix "$subpkgdir"/usr/bin/$file
 	done
-	mv "$pkgdir"/etc/$pkgname/pear.conf "$subpkgdir"/etc/$pkgname/ || return 1
-	mv "$pkgdir"/usr/share "$subpkgdir"/usr/
+
+	_mv etc/$pkgname/pear.conf "$subpkgdir"/etc/$pkgname/
+	_mv usr/share "$subpkgdir"/usr/
 }
 
 common() {
 	pkgdesc="$pkgdesc (common config)"
 	depends=""
 
-	mkdir -p "$subpkgdir"/usr
-	mv "$pkgdir"/usr/lib "$subpkgdir"/usr/ || return 1
-	mv "$pkgdir"/etc "$subpkgdir"/ || return 1
+	cd "$pkgdir"
 
-	mkdir "$subpkgdir"/etc/$pkgname/conf.d
+	_mv usr/lib "$subpkgdir"/usr/
+	_mv etc "$subpkgdir"/
+	mkdir -p "$subpkgdir"/$_extension_confd
 }
 
 phar() {
-	_package_ext || return 1
+	_extension
+
+	cd "$pkgdir"
+
 	mkdir -p "$subpkgdir"/usr/bin
-	mv "$pkgdir"/usr/bin/phar* "$subpkgdir"/usr/bin/
+	mv usr/bin/phar.phar "$subpkgdir"/usr/bin/phar.phar$_suffix
+
+	rm usr/bin/phar
+	ln -s phar.phar$_suffix "$subpkgdir"/usr/bin/phar$_suffix
+
+	if [ "$_default_php" = yes ]; then
+		ln -s phar.phar$_suffix "$subpkgdir"/usr/bin/phar.phar
+		ln -s phar.phar$_suffix "$subpkgdir"/usr/bin/phar
+	fi
 }
 
-_package_ext() {
+_extension() {
 	local extname="${subpkgname#$pkgname-}"
-	local extdepends="$(eval "echo \$depends_$extname")"
-	pkgdesc="PHP7 extension: $extname"
-	depends="$depends $extdepends"
+	local extdepends="$(eval "echo \$_depends_$extname")"
+	pkgdesc="PHP$_suffix extension: $extname"
```

Comment:
discussed briefly at IRC
IMO better to have [bundled](http://php.net/manual/en/extensions.membership.php#extensions.membership.bundled) in some meta package like `php-bundled` but most of distros using `php-common` with a broader set of extensions